### DISCUSSION TITLE
SO 2.3.130 Error in SOC Hunt view

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#8104] [Unsupported Versions] [answered]
``Hello!

I have recently updated to SO 2.3.130 and now I get the following alert in the Hunt view of SOC:

![image](https://user-images.githubusercontent.com/81571847/173018053-eb7185cd-2eac-4681-8a96-21f416174632.png)

I checked the sensoroni-server.log and found the following error whenever I try to hunt for something:

```
timestamp=2022-06-10T07:00:06.771017766Z level=info message="Searching Elasticsearch" query="{\"aggs\":{\"bottom\":{\"terms\":{\"field\":\"event.module.keyword\",\"order\":{\"_count\":\"asc\"},\"size\":500}},\"groupby_0|event.module\":{\"aggs\":{\"groupby_0|event.module|event.dataset\":{\"terms\":{\"field\":\"event.dataset.keyword\",\"order\":{\"_count\":\"desc\"},\"size\":500}}},\"terms\":{\"field\":\"event.module.keyword\",\"order\":{\"_count\":\"desc\"},\"size\":500}},\"timeline\":{\"date_histogram\":{\"field\":\"@timestamp\",\"fixed_interval\":\"1h\",\"min_doc_count\":1}}},\"query\":{\"bool\":{\"filter\":[],\"must\":[{\"query_string\":{\"analyze_wildcard\":true,\"default_field\":\"*\",\"query\":\"(\\\"Executable in ADS\\\") AND NOT _index: \\\"*:so-case*\\\"\"}},{\"range\":{\"@timestamp\":{\"format\":\"strict_date_optional_time\",\"gte\":\"2022-06-09T09:00:06+02:00\",\"lte\":\"2022-06-10T09:00:06+02:00\"}}}],\"must_not\":[],\"should\":[]}},\"size\":500}" requestId=e1c7f0bd-959d-4a4d-afcf-46278bdb10bf
timestamp=2022-06-10T07:00:07.057255125Z level=warn message="Shard failure" reason="failed to create query: field expansion for [*] matches too many fields, limit: 3500, got: 3950" type=query_shard_exception
timestamp=2022-06-10T07:00:07.057442664Z level=warn message="Request did not complete successfully" error=ERROR_QUERY_FAILED_ELASTICSEARCH requestId=e1c7f0bd-959d-4a4d-afcf-46278bdb10bf requestor=14ab1b02-d6b4-4aa5-a0bc-7a0ab233a19a
```

Any ideas how to fix this and why I suddenly reach this field limit? Thanks in advance!

Cheers, Ben

**Comment by presianbg** [comment]
I think I had exact same error on the new `Cases` view. 
Not sure if the update process just needed more time to complete or clearing the browser cache/cookies helped, but I have no such error anymore.

**Comment by Jaap79** [comment]
Same error here. I increased the max_clause_count to just above the required number but my search nodes are now falling apart...

**Comment by YannickZ5** [comment]
I have also the same error.

**Comment by tiger-greenhell** [comment]
Release Notes
2.3.130 Changes
......
FIX: Ensure failed elastic queries show an error on the SOC UI [#7846](https://github.com/Security-Onion-Solutions/securityonion/issues/7846)

Currently if there's a complete failure of the query then the SOC UI shows as if there were no results.

Example of errors that should be visible on the SOC UI:

    shards unavailable
    field count exceeded
    query clauses exceeded
    etc

**Reply by ben-sec** [reply]
OK, but how can I fix the root cause of the error in order to use the hunt view again?

**Comment by dougburks** [comment, marked as answer]
> timestamp=2022-06-10T07:00:07.057255125Z level=warn message="Shard failure" reason="failed to create query: field expansion for [*] matches too many fields, limit: 3500, got: 3950" type=query_shard_exception

https://docs.securityonion.net/en/2.3/elasticsearch.html#field-expansion-matches-too-many-fields

**Reply by ben-sec** [reply]
Hi Doug,

I put 
```
config:
    indices.query.bool.max_clause_count: 4000
```
in `/opt/so/saltstack/local/pillar/minions/my_searchnode.sls`, did a highstate and restarted ES on my search node without success. Wrong place for the config change? According to the error message it's still at 3500 instead of 4000, but on my search node I find in `/opt/so/conf/elasticsearch/elasticsearch.yml` the correct setting `indices.query.bool.max_clause_count: 4000`.

Cheers, Ben

**Reply by ben-sec** [reply]
Now I put the `indices.query.bool.max_clause_count: 4000` setting in global.sls and it seems to work and so far my search node is working well.

**Reply by Jaap79** [reply]
Hi Ben-Sec,

I tried that as well but performance is dreadful if I up it to the minimum requirement (currently at 4500 else my Hunt / searches simply won't work most of the time). 

It seems though, I ran a quick glance through the indices, that there's a bunch of so-playbook-alerts-<date> indices with a huge amount of fields. A random one I picked easily exceeds 2600 fields, I can imagine there must be an index in there with >4000 fields but haven't found it yet. I'm not too well versed in Elastic so I can't easily figure out a way to see which indexes have how many fields..

**Reply by tempest3991** [reply]
> Now I put the `indices.query.bool.max_clause_count: 4000` setting in global.sls and it seems to work and so far my search node is working well.

Where did you put this in the global.sls? I added it under elasticsearch, but it doesn't seem to be helping

**Reply by ben-sec** [reply]
```
elasticsearch:
   config:
       indices.query.bool.max_clause_count: 4000
```

**Reply by Jaap79** [reply]
@tempest3991 :

I put it in /opt/so/saltstack/local/salt/elasticsearch/defaults.yaml (on the *manager node*):
elasticsearch:
[...]
  indices:
      query:  
        bool:
          max_clause_count: 4500

then call a highstate:
salt (nodename)_searchnode state.highstate

And then it worked.

**Reply by dougburks** [reply]
The Elasticsearch customization section of our documentation has been updated to address this:
https://docs.securityonion.net/en/2.3/elasticsearch.html#customization

**Reply by ben-sec** [reply]
Thanks Doug!
